harness/src/main/java/org/jboss/pvt/harness/validators/JDKCompatibleCheckValidator.java

```diff
@@ -0,0 +1,83 @@
+package org.jboss.pvt.harness.validators;
+
+import org.jboss.pvt.harness.configuration.PVTConfiguration;
+import org.jboss.pvt.harness.exception.PVTException;
+import org.jboss.pvt.harness.utils.ClassVersion;
+import org.jboss.pvt.harness.utils.ClassVersionInspector;
+import org.jboss.pvt.harness.utils.DirUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * @author <a href="mailto:yyang@redhat.com">Yong Yang</a>
+ */
+public class JDKCompatibleCheckValidator implements Validator {
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Override
+    public boolean validate(PVTConfiguration pvtConfiguration) throws PVTException {
+        File eapDir = pvtConfiguration.getDistributionDirectory();
+
+        Collection<File> jarFiles = DirUtils.listFilesRecursively( eapDir, new FileFilter() {
+            public boolean accept(File pathname) {
+                return  pathname.isFile() && pathname.getName().endsWith(".jar");
+            }
+        });
+
+        logger.info("Jars: " + Arrays.toString(jarFiles.toArray()));
+
+
+        List<File> failedJars = new ArrayList<>();
+
+        for (File jarFile : jarFiles) {
+            if (jarFile.isFile()) {
+
+                if (isFilter(pvtConfiguration, jarFile)) {
+                    logger.warn("Excluding Jar, ", jarFile);
+                    continue;
+                }
+
+                if(!ClassVersionInspector.checkJarVersion(jarFile.toString(), getVersion(pvtConfiguration)[0], getVersion(pvtConfiguration)[1])) {
+                    failedJars.add(jarFile);
+                }
+            }
+        }
+        if(!failedJars.isEmpty()) {
+            logger.error("JDK compatible check failed: ", Arrays.toString(failedJars.toArray()));
+            return false;
+        }
+        return true;
+    }
+
+    protected List<String> getFilters(PVTConfiguration pvtConfiguration){
+        return pvtConfiguration.getTestCase(this.getClass().getName()).getFilters();
+    }
+
+    protected boolean isFilter(PVTConfiguration pvtConfiguration, File jarFile){
+        List<String> filters = getFilters(pvtConfiguration);
+        for(String filter : filters){
+            if(Pattern.compile(filter).matcher(jarFile.getAbsoluteFile().getName()).matches()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * get expected class version scope
+     * @param pvtConfiguration
+     * @return
+     */
+    protected ClassVersion[] getVersion(PVTConfiguration pvtConfiguration){
+        return new ClassVersion[]{ClassVersion.JAVA_0, ClassVersion.JAVA_18};
```

Comment:
If I understand correctly, @yongyang is expecting to return a scope [minmal support jdkversion, maximum support jdk version] instead of only maximum versions.
For example, a jdk 1.1 compiled  jar would pass the validator for <= jdk1.6 check
But a jdk 1.1 compiled jar wouldn't pass a validaotr for checking [jdk1.5,jdk1.6] check

This makes sense to me to have a scope to return.  
Nick has a good point. there is no JAVA_0 matched JDK version. 
Having a seach on JDK version mapper:
Target   Major.minor
1.2      46.0
1.3      47.0
1.4      48.0
5 (1.5)  49.0
6 (1.6)  50.0
7 (1.7)  51.0
8 (1.8)  52.0

@yongyang is there any other reason why we need a fake value for JAVA_0?  can we start the minmal JDK version from JAVA_12?